sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -364,3 +373,229 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def test_mice_pipeline_grid_search():
+    # Test imputation within a pipeline + gridsearch.
+    pipeline = Pipeline([('imputer', MICEImputer(missing_values=0,
+                                                 n_imputations=1,
+                                                 n_burn_in=1,
+                                                 random_state=0)),
+                         ('tree', tree.DecisionTreeRegressor(random_state=0))])
+
+    parameters = {
+        'imputer__initial_strategy': ["mean", "median", "most_frequent"]
+    }
+
+    n = 100
+    d = 10
+    X = sparse_random_matrix(n, d, density=0.50).toarray()
+    Y = np.random.random((n, d))
+    gs = GridSearchCV(pipeline, parameters)
+    gs.fit(X, Y)
+
+
+def test_mice_rank_one():
+    np.random.seed(0)
```

Comment:
we try to avoid this to ensure thread safety. Use rng = np.random.RandomState() and pass it around instead

Reply:
Gotcha. I'll change that everywhere.